### ISSUE TITLE
Build installer tests against the live ref pack

### ISSUE THREAD

**Issue by jkoritzinsky** [dotnet/runtime#45972]
We discovered in #45781 that the installer test assets build against a pre-built ref pack.

Since they build at the end of the build and we want to build them as though they were built against the released product, we should change their build to build against the live ref pack. This way we'll ensure that each test is testing what it should without including any extra roll-forward behaviors that we may accidentally start depending on.

**Comment by NikolaMilosavljevic** [comment]
[Triage] This should only consume live ref-packs for new release and not in servicing, as those are not reshipped.

**Comment by NikolaMilosavljevic** [comment]
[Triage] @ViktorHofer is this still needed?

**Comment by ViktorHofer** [comment]
Yes that's afaik they still don't compile against the live targeting pack.

**Comment by ViktorHofer** [comment]
This is again causing issues when upgrading the TFM (from net7.0 to net8.0): https://github.com/dotnet/runtime/pull/78354/. Will again work-around it for now but we should really fix this.
